src/function/gds/output_writer.cpp

```diff
@@ -19,32 +19,35 @@ void PathsOutputs::beginWritingOutputsForDstNodesInTable(common::table_id_t tabl
     bfsGraph.pinNodeTable(tableID);
 }
 
+static std::unique_ptr<ValueVector> createVector(const LogicalType& type,
+    storage::MemoryManager* mm, std::vector<ValueVector*>& vectors) {
+    auto vector = std::make_unique<ValueVector>(type.copy(), mm);
+    vector->state = DataChunkState::getSingleValueDataChunkState();
+    vectors.push_back(vector.get());
+    return vector;
+}
+
 RJOutputWriter::RJOutputWriter(main::ClientContext* context, RJOutputs* rjOutputs)
     : context{context}, rjOutputs{rjOutputs} {
     auto mm = context->getMemoryManager();
-    srcNodeIDVector = std::make_unique<ValueVector>(LogicalType::INTERNAL_ID(), mm);
-    dstNodeIDVector = std::make_unique<ValueVector>(LogicalType::INTERNAL_ID(), mm);
-    srcNodeIDVector->state = DataChunkState::getSingleValueDataChunkState();
+    srcNodeIDVector = createVector(LogicalType::INTERNAL_ID(), mm, vectors);
+    dstNodeIDVector = createVector(LogicalType::INTERNAL_ID(), mm, vectors);
     srcNodeIDVector->setValue<nodeID_t>(0, rjOutputs->sourceNodeID);
-    dstNodeIDVector->state = DataChunkState::getSingleValueDataChunkState();
-    vectors.push_back(srcNodeIDVector.get());
-    vectors.push_back(dstNodeIDVector.get());
 }
 
 PathsOutputWriter::PathsOutputWriter(main::ClientContext* context, RJOutputs* rjOutputs,
```

Comment:
I would consider simplifying this by always writing the direction now. So always create a direction value vector and always write it and the outputter can ignore it.

Reply:
I'll think about it. Right now it's not too much work yet. Later if I observe we need to do if check in multiple places I might apply this change.